app/src/db/projects.ts

```diff
@@ -411,14 +411,19 @@ export async function removeProjectOrganization({
 }
 
 export async function deleteProject({ id }: { id: string }) {
-  return prisma.project.update({
-    where: {
-      id,
-    },
-    data: {
-      deletedAt: new Date(),
-    },
-  })
+  return Promise.all([
+    prisma.project.update({
+      where: {
+        id,
+      },
+      data: {
+        deletedAt: new Date(),
+      },
+    }),
+    prisma.projectRepository.deleteMany({
+      where: { projectId: id },
+    }),
```

Comment:
let's make sure that deleting the project succeeds before we delete the repos

Reply:
You can use prisma transactions to ensure execution of all transactions with [prisma transactions](https://www.prisma.io/docs/orm/prisma-client/queries/transactions#sequential-prisma-client-operations)

> Prisma Client will ensure that either all three create operations succeed or none of them succeed.

something along these lines should work
```suggestion
    return prisma.$transaction(async (tx) => {
      await tx.project.update({
        where: {
          id,
        },
        data: {
          deletedAt: new Date(),
        },
      });
      await tx.projectRepository.deleteMany({
        where: { projectId: id },
      });
    });
```

Reply:
Thanks @nikolagigic. Went with your approach; appreciate the clean solution!